drivers/SmartThings/zigbee-switch/src/bad_on_off_data_type/init.lua

```diff
@@ -0,0 +1,42 @@
+-- Copyright 2023 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local zcl_clusters = require "st.zigbee.zcl.clusters"
+local data_types = require "st.zigbee.data_types"
+local capabilities = require "st.capabilities"
+
+-- There are reports of at least one device (SONOFF 01MINIZB) which occasionally
+-- reports this value as an Int8, rather than a Boolean, as per the spec
+local function incorrect_data_type_detected(opts, driver, device, zb_rx, ...)
+  return opts.dispatcher_class == "ZigbeeMessageDispatcher" and zb_rx.body.zcl_body.attr_record.data_type ~= data_types.Boolean.ID
```

Comment:
This can_handle can be called on any zigbee message, so there is no guarantee that `attr_record` will exist, so this could cause an error.  And _if_ we are going to compare against 0 explicitly below, we might want to test against specifically INT and UINT types here instead of just _not_ boolean.  My only hesitation with this approach is that this will catch _all_ non boolean messages from any device even though we are trying to handle a specific device.  But it might still be worth considering to avoid re-fixing for other devices.

Reply:
It doesn't look like any other sub-drivers override our handling for this attribute.